client.go

```diff
@@ -98,24 +98,25 @@ func (c *Client) doRequest(r *http.Request) (*http.Response, error) {
 	if err != nil {
 		return nil, err
 	}
-	defer resp.Body.Close()
 
 	if resp.StatusCode < 200 || resp.StatusCode > 299 {
-		return nil, &httpError{
-			status:     resp.Status,
-			statusCode: resp.StatusCode,
-		}
+		return nil, NewHTTPError(resp)
 	}
 
 	var aer autheticationErrorResponse
 
 	// We read and save the response body so that if we don't have error messages
 	// we can set it again for future usage
+	defer resp.Body.Close()
```

Comment:
Why did you move it down here? We'd still want it to be handled before the previous conditional. If the status code is anything else, we now won't close the body.

Reply:
woops - good catch.  I meant to add `resp.Body.Close()` to `NewHTTPError`.  I moved this down here so the function won't try to close the Body twice.  I moved it down to one line above where we actually try to interact with the Body.

Reply:
I decided to just move it back and not add it to NewHTTPError so it's done in one place.